File: src/api/services/roleService.js
import apiClient from '@/config/axios';

const BASE = '/roles';

/**
 * Role Management API service.
 * Wraps all calls to /api/v1/roles/* endpoints.
 */
const roleService = {
  /** List all active roles. */
  getAllActive: () => apiClient.get(BASE),

  /** Get a single role by ID. */
  getById: (id) => apiClient.get(`${BASE}/${id}`),

  /** Replace all roles on a user. */
  assignRoles: (userId, roleIds) =>
    apiClient.put(`${BASE}/user/${userId}`, roleIds),

  /** Add a single role to a user. */
  addRole: (userId, roleId) =>
    apiClient.post(`${BASE}/user/${userId}/role/${roleId}`),

  /** Remove a single role from a user. */
  removeRole: (userId, roleId) =>
    apiClient.delete(`${BASE}/user/${userId}/role/${roleId}`),
};

export default roleService;

File: backend/src/main/java/com/smartwms/service/RoleService.java
package com.smartwms.service;

import com.smartwms.dto.response.RoleResponse;

import java.util.List;

/**
 * Service contract for Role management operations.
 */
public interface RoleService {

    /** List all active roles. */
    List<RoleResponse> getAllActive();

    /** Get a single role by ID. */
    RoleResponse getById(Long id);
}
